src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateCopy = { ...state };
+  const result = [];
+
+  for (const obj of actions) {
```

Comment:
```suggestion
  for (const action of actions) {
```
use descriptive names

Reply:
also use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) for usage of type, extraData, keysToRemove